### ISSUE TITLE
Compatibility with old versions

### ISSUE THREAD

**Issue by Lolson16** [biomodhub/biomod2#144]
Hello,

I used biomod2 (v 3.3-7) to make an ensemble SDM for my species of interest a couple years ago, and everything worked great. I then used the BIOMOD_EnsembleForecasting command to create future predictions of my model and that worked too. Now, though, I'm using the new version of biomod2 (v 4.1-2) and I'd like to redo my future predictions from my old ensemble models, but when I try the same code, it fails. 

I'm able to load the old 'ensemble.models.out' file but when I try to run BIOMOD_EnsembleForecasting, I get the error:

"Error in .fun_testIfInherits(TRUE, "bm.em", bm.em, "BIOMOD.ensemble.models.out") : 
  
bm.em must be a 'BIOMOD.ensemble.models.out' object"

Does this just mean that my old files aren't compatible with the new version of biomod2? I also tried reinstalling the older version (v 3.3-7) but I can't get the Maxent dependencies to load so it won't install (here I get the error  'Warning in system(cmd) : 'make' not found  ERROR: compilation failed for package 'maxent').

Any thoughts would be greatly appreciated! Please let me know if you need more information.

Here is my sessionInfo()
R version 4.1.3 (2022-03-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] raster_3.5-15 rgdal_1.5-32  sp_1.4-7      biomod2_4.1-2

loaded via a namespace (and not attached):
 [1] gbm_2.1.8.1            tidyselect_1.1.2       terra_1.5-21          
 [4] purrr_0.3.4            reshape2_1.4.4         splines_4.1.3         
 [7] lattice_0.20-45        colorspace_2.0-3       vctrs_0.4.1           
[10] generics_0.1.2         utf8_1.2.2             survival_3.2-13       
[13] rlang_1.0.2            pillar_1.7.0           glue_1.6.2            
[16] plotmo_3.6.2           foreach_1.5.2          lifecycle_1.0.1       
[19] plyr_1.8.7             mda_0.5-3              stringr_1.4.0         
[22] munsell_0.5.0          gtable_0.3.0           codetools_0.2-18      
[25] maxnet_0.1.4           class_7.3-20           fansi_1.0.3           
[28] Rcpp_1.0.8.3           scales_1.2.0           plotrix_3.8-2         
[31] abind_1.4-5            TeachingDemos_2.12     ggplot2_3.3.6         
[34] stringi_1.7.6          dplyr_1.0.9            grid_4.1.3            
[37] cli_3.3.0              tools_4.1.3            magrittr_2.0.3        
[40] PresenceAbsence_1.1.10 tibble_3.1.7           randomForest_4.7-1.1  
[43] Formula_1.2-4          crayon_1.5.1           pkgconfig_2.0.3       
[46] MASS_7.3-55            ellipsis_0.3.2         Matrix_1.4-0          
[49] data.table_1.14.2      pROC_1.18.0            reshape_0.8.9         
[52] iterators_1.0.14       rpart_4.1.16           earth_5.3.1           
[55] R6_2.5.1               nnet_7.3-17            compiler_4.1.3        

And here is the summary of the ensemble model I'm trying to work with:
> em_all
An object of class "BIOMOD.EnsembleModeling.out"
<S4 Type Object>
attr(,"sp.name")
[1] "Run31"
attr(,"expl.var.names")
 [1] "comp_topo_ind2"                 "heat_loadp"                    
 [3] "int_moisturep"                  "mcmt250"                       
 [5] "MinDensity_mean_2004_2017_250"  "PHIHOX_M_sl2_250m_ll_clip_proj"
 [7] "ppt_sm250"                      "ppt_wt250"                     
 [9] "rh250"                          "surf_areap"                    
[11] "SWE_mean_2004_2017_250"         "tpi10kp"                       
attr(,"models.out.obj")
An object of class "BIOMOD.stored.models.out"
Slot "val":
NULL

Slot "inMemory":
[1] FALSE

Slot "link":
[1] "Run31/Run31.ID31.models.out"

attr(,"eval.metric")
[1] ""
attr(,"eval.metric.quality.threshold")
[1] 0
attr(,"em.models")
attr(,"em.models")$Run31_EMmeanByROC_mergedAlgo_mergedRun_mergedData

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-= biomod2_model -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=

         model name : Run31_EMmeanByROC_mergedAlgo_mergedRun_mergedData

         model class : EMmean

         This model doesn't have its own scale

Error in cat("\n\t modeling folder:", object@dir_name, fill = .Options$width) : 
  no slot of name "dir_name" for this object of class "EMmean_biomod2_model"
>
And here's the code I'm trying to run, and the error it generates: 

projd.ens.mods <-BIOMOD_EnsembleForecasting( bm.em=em_all,
                            bm.proj = NULL,
                            new.env = r,
                            new.env.xy = NULL,
                            models.chosen = 'all',
                            proj.name = proj_namer,
                            metric.binaary = NULL,
                            metric.filter = NULL,
                            compress = TRUE)

-=-=-=-=-=-=-=-=-=-=-=-= Do Ensemble Models Projection -=-=-=-=-=-=-=-=-=-=-=-=
Error in .fun_testIfInherits(TRUE, "bm.em", bm.em, "BIOMOD.ensemble.models.out") : 
  
bm.em must be a 'BIOMOD.ensemble.models.out' object

**Comment by rpatin** [comment]
Hi @Lolson16 
Thank you for reporting :pray: and sorry for the inconvenience.

Unfortunately there are no back compatibility between current 4.1-2 version and 3.3-7 version (which date back to 2016) so that would be quite difficult to use your old results with current `biomod2` version.

I think you have two different options:
1. re-run your models with current version (given that you have all data and modeling steps available)
2. install version 3.3-7, which will require to install a series of outdated package. Concerning the error you encounter with this installation, I suppose you are missing Rtools on your system: https://cran.r-project.org/bin/windows/Rtools/

Best,
Rémi

**Comment by Lolson16** [comment]
Hi Rémi,
Thank you! It's helpful to know that so at least I know there's nothing I need to fix. I was able to get an old version installed and you were right, I did need to install Rtools, and to install archived verions of 'terra' and some other dependencies. I never did get an old version of Maxent to install, even with Rtools it always gave me an error, but I somehow got a workaround by copying the Maxent folder and its contents into my library, and that seemed to be enough to get the old version of biomod to run ensemble predictions. 

Thanks very much for your reply,
Lucretia
